include/bayes_tracking/multitracker.h

```diff
@@ -230,6 +236,17 @@ class MultiTracker {
         m_filters[j].filter->predict_observation(om, zp, Zp);
         S = Zp + om.Z;  // H*P*H' + R
         for (int i = 0; i < M; i++) {
+
+         // Only Check NN, NNJPDA tag associate not yet implemented
+         if (alg == NN) 
+         {
+           // Assign maximum cost if observations and trajectories labelled do not match
+           if (m_observations[i].tag != m_filters[j].tag) 
```

Comment:
This means that whenever `NN` is used, the tag checking will be employed, right? That's a change to the default behaviour that present to date. I think, it should be an optional behaviour. I suggest to actually implement this as another association algorithm, e.g. `NN_TAGGED` or `NN_LABELED` and making sure the original one is left as is.